File: Introduction to Object-Oriented Programming/Computer Cluster/regneklynge.py
from racks import Rack

class Regneklynge:

    def __init__(self, noderPerRack):
        self._noderPerRack = noderPerRack
        self._regneklynge = []

    def settInnNode(self, node):
        telling = len(self._regneklynge)-1
        if len(self._regneklynge) == 0:
            nyrack = Rack()
            nyrack.settInn(node)
            self._regneklynge.append(nyrack)
        else:
            tellendeRack = self._regneklynge[telling]
            if tellendeRack.getAntNoder() == self._noderPerRack:
                nyereRack = Rack()
                nyereRack.settInn(node)
                self._regneklynge.append(nyereRack)
            else:
                tellendeRack.settInn(node)

    def antProsessorer(self):
        antall = 0
        for rack in self._regneklynge:
            antall += rack.antProsessorer()
        return antall

    def noderMedNokMinne(self, paakrevdMinne):
        antall = 0
        for rack in self._regneklynge:
            antall += rack.noderMedNokMinne(paakrevdMinne)
        return antall

    def antRacks(self):
        antall = len(self._regneklynge)
        return antall
        
    def daniel(self):
    

File: Object Oriented Programming/Computer Cluster/Regneklynge.java
import java.util.ArrayList;

class Regneklynge{
  private int noderPerRack;
  private ArrayList<Racks> rackliste = new ArrayList<Racks>();

  public Regneklynge(int maxNoder){
    noderPerRack = maxNoder;
  }
  public void settInnNode(Node noden){
    int teller = (rackliste.size()) -1;
    if (rackliste.size() == 0){
      Racks nyRack = new Racks();
      nyRack.settInn(noden);
      rackliste.add(nyRack);
    }
    else{
      Racks tellendeRack = rackliste.get(teller);
      if(tellendeRack.getAntNoder() == noderPerRack){
        Racks nyereRack = new Racks();
        nyereRack.settInn(noden);
        rackliste.add(nyereRack);

      }
      else{
        tellendeRack.settInn(noden);
      }

    }

  }

  public int antProsessorer(){
    int antall = 0;
    for(Racks rack: rackliste){
      antall += rack.antProsessorer();
    }
    return antall;
  }

  public int noderMedNokMinne(int paakrevdMinne){
    int antall = 0;
    for(Racks rack: rackliste){
      antall += rack.noderMedNokMinne(paakrevdMinne);

    }
    return antall;
  }

  public int antRacks(){
    return rackliste.size();
  }

}
